crutest/src/main.rs

```diff
@@ -2431,15 +2471,15 @@ async fn replace_while_reconcile(
                 wc.ds_count,
                 wc.active_count
             );
-            if wc.up_count + wc.ds_count == 0 && wc.active_count == 3 {
-                info!(log, "[{c}] Replay: All jobs finished, all DS active.");
+            if wc.up_count + wc.ds_count == 0 && wc.active_count == ds_total {
+                info!(log, "[{c}] All jobs finished, all DS active.");
                 break;
             }
             tokio::time::sleep(tokio::time::Duration::from_secs(4)).await;
         }
 
-        old_ds = (old_ds + 1) % 4;
-        new_ds = (new_ds + 1) % 4;
+        old_ds = (old_ds + 1) % (ds_total as u32 + 1);
```

Comment:
Nit: what's with the `as u32` cast here (and below)?

Reply:
I'm just doing what the compiler tells me, man!

`old_ds` which is sent to dsc_client functions wants it to be u32.
However, when we use it as an index for `targets[`,  both `old_ds` and `new_ds` need to be usize.

If you think it's better, I could make it a `usize`, and cast to` u32` when I send it to dsc?